File: src/app/model/TemperatureMetric.ts
import { MetricService } from "../services/metric.service";
import { Metric, MetricType } from "./Metric";

export class TemperatureMetric extends Metric {
    constructor(metricService: MetricService) {
        super(MetricType.TEMPERATURE, 'Temperature', '°K', metricService, '°C');
    }

    startLogging(): void {

    }

    stopLogging(): void {

    }
}

File: android/app/src/main/java/com/openbikecomputer/model/TemperatureMetric.java
package com.openbikecomputer.model;

import com.openbikecomputer.service.IMetricService;

public class TemperatureMetric extends Metric {
    public TemperatureMetric(IMetricService metricService) {
        super(MetricType.TEMPERATURE, "Temperature", "\u00b0K", metricService, "\u00b0C");
    }

    @Override
    public void startLogging() {
    }

    @Override
    public void stopLogging() {
    }
}
